buildconfig/CMake/VersionNumber.cmake

```diff
@@ -1,7 +1,20 @@
 # Set the version number here for MantidVersion and the package filenames This follows Semantic Versioning
 # https://semver.org
-set(VERSION_MAJOR 6)
-set(VERSION_MINOR 2)
+if(CONDA_BUILD)
+  set(_conda_package_version $ENV{PKG_VERSION})
+  string(REPLACE "." ";" _conda_package_version ${_conda_package_version})
+  set(_conda_package_list ${_conda_package_version})
+
+  list(GET _conda_package_version 0 VERSION_MAJOR)
+  list(GET _conda_package_version 1 VERSION_MINOR)
+  list(SUBLIST _conda_package_list 2 2 _version_patch_list)
+  set(VERSION_PATCH)
+  string(REPLACE ";" "." VERSION_PATCH "${_version_patch_list}")
+  message(STATUS "Version: ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}")
```

Comment:
This is at present decomposing the version numbers in the env variable PKG_VERSION which is set in the conda recipe.
https://github.com/mantidproject/conda-recipes/blob/75bb15030b01fa8801deb2184c69392c5b19d011/recipes/mantid/meta.yaml#L1
This line can be set in a few ways, manually, using whatever bot or automatic process to update it. Right now the main way will be a script in the nightly pipeline that updates the version based on the numbers on lines 15 and 16 in VersionNumber.cmake of this PR (specifically it greps the file for a line that contains `set(VERSION_MAJOR` and `set(VERSION_MINOR`). Right now the way we set version major and minor is just by setting them in this file.

Reply:
Okay, can we put a comment in the Conda section stating something like:

`# This will automatically update from the Conda recipe which looks at VERSION_MAJOR and VERSION_MINOR below`

So future technical release managers don't go on a wild goose chase looking for how to "update Conda's version too"

Reply:
Ah yeah we just kinda do it automatically, I can add a comment yes.